terraform import azurerm_virtual_machine results in panic error

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

```
[coopem67@localhost bi-virtual-machines]$ terraform -v
Terraform v0.11.10
+ provider.azurerm v1.19.0
```

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* azurerm_virtual_machine

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please note I made my Azure subscription ID all zeros; I had the correct ID in the command I ran.
```hcl
terraform import -var-file=../storage.key -var-file=../tfvars/bi-prod.tfvars azurerm_virtual_machine.main[0] /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/BusinessIntelligence/providers/microsoft.compute/VirtualMachines/biprodintrunvm01
```

### Debug Output

[Debug gist](https://gist.githubusercontent.com/mlcooper/a753bdd928d364f595811c200ff9c1b1/raw/d67f7bdbdfcbc6e8522c346259f76ca2315deedc/gistfile1.txt)

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

```
panic: runtime error: invalid memory address or nil pointer dereference
2018-11-19T14:14:53.080-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1cc194d]
2018-11-19T14:14:53.080-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: 
2018-11-19T14:14:53.080-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: goroutine 46 [running]:
2018-11-19T14:14:53.080-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceArmVirtualMachineRead(0xc000528700, 0x2156020, 0xc0005be000, 0xc000528700, 0x0)
2018-11-19T14:14:53.080-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_virtual_machine.go:732 +0x5bd
2018-11-19T14:14:53.081-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc0003878f0, 0xc0004101e0, 0x2156020, 0xc0005be000, 0xc0003aca78, 0x4c2201, 0x1dfbba0)
2018-11-19T14:14:53.081-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:352 +0x160
2018-11-19T14:14:53.081-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc00039e930, 0xc000410190, 0xc0004101e0, 0xc000588a80, 0x18, 0x7f670a9c5000)
2018-11-19T14:14:53.081-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:308 +0x92
2018-11-19T14:14:53.081-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc00041dc80, 0xc000713440, 0xc0007136c0, 0x0, 0x0)
2018-11-19T14:14:53.082-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:549 +0x4e
2018-11-19T14:14:53.082-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: reflect.Value.call(0xc0002d38c0, 0xc00000d160, 0x13, 0x218cf71, 0x4, 0xc000928f18, 0x3, 0x3, 0xc0004a5440, 0x532061696c617274, ...)
2018-11-19T14:14:53.082-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: 	/opt/goenv/versions/1.11.1/src/reflect/value.go:447 +0x449
2018-11-19T14:14:53.083-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: reflect.Value.Call(0xc0002d38c0, 0xc00000d160, 0x13, 0xc000800718, 0x3, 0x3, 0x6145206164616e61, 0x736557222c227473, 0x2c22322053552074)
2018-11-19T14:14:53.083-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: 	/opt/goenv/versions/1.11.1/src/reflect/value.go:308 +0xa4
2018-11-19T14:14:53.083-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: net/rpc.(*service).call(0xc0004a5740, 0xc000410f00, 0xc0003620f0, 0xc000362110, 0xc0000c6800, 0xc000123ce0, 0x1dfbb60, 0xc000713440, 0x16, 0x1dfbba0, ...)
2018-11-19T14:14:53.084-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: 	/opt/goenv/versions/1.11.1/src/net/rpc/server.go:384 +0x14e
2018-11-19T14:14:53.084-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: created by net/rpc.(*Server).ServeCodec
2018-11-19T14:14:53.084-0500 [DEBUG] plugin.terraform-provider-azurerm_v1.19.0_x4: 	/opt/goenv/versions/1.11.1/src/net/rpc/server.go:481 +0x47e
2018/11/19 14:14:53 [ERROR] root: eval: *terraform.EvalRefresh, err: azurerm_virtual_machine.main: unexpected EOF
2018/11/19 14:14:53 [ERROR] root: eval: *terraform.EvalSequence, err: azurerm_virtual_machine.main: unexpected EOF
```

### Expected Behavior

I expected that terraform would import this first of two Azure VM's

### Actual Behavior

```
Error: azurerm_virtual_machine.main[0] (import id: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/BusinessIntelligence/providers/microsoft.compute/VirtualMachines/biprodintrunvm01): 1 error(s) occurred:

* import azurerm_virtual_machine.main[0] result: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/BusinessIntelligence/providers/microsoft.compute/VirtualMachines/biprodintrunvm01: azurerm_virtual_machine.main: unexpected EOF
```
See debug gist

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

```hcl
terraform import -var-file=../storage.key -var-file=../tfvars/bi-prod.tfvars azurerm_virtual_machine.main[0] /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/BusinessIntelligence/providers/microsoft.compute/VirtualMachines/biprodintrunvm01
```

### Important Factoids

I'm running in `West US`.  This VM is one of two in an Azure Availability Set.  There is no auto-scaling of any type configured.

<!--- Are there anything atypical about your accounts that we should know? For example: Running in a Azure China/Germany/Government? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->

* #0000

--- a/azurerm/resource_arm_virtual_machine.go
+++ b/azurerm/resource_arm_virtual_machine.go
@@ -721,91 +721,90 @@ func resourceArmVirtualMachineRead(d *schema.ResourceData, meta interface{}) err
 		d.Set("location", azureRMNormalizeLocation(*location))
 	}
 
-	if resp.Plan != nil {
-		if err := d.Set("plan", flattenAzureRmVirtualMachinePlan(resp.Plan)); err != nil {
-			return fmt.Errorf("[DEBUG] Error setting Virtual Machine Plan error: %#v", err)
-		}
+	if err := d.Set("plan", flattenAzureRmVirtualMachinePlan(resp.Plan)); err != nil {
+		return fmt.Errorf("Error setting `plan`: %#v", err)
 	}
 
-	d.Set("identity", flattenAzureRmVirtualMachineIdentity(resp.Identity))
-
-	if resp.VirtualMachineProperties.AvailabilitySet != nil {
-		d.Set("availability_set_id", strings.ToLower(*resp.VirtualMachineProperties.AvailabilitySet.ID))
+	if err := d.Set("identity", flattenAzureRmVirtualMachineIdentity(resp.Identity)); err != nil {
+		return fmt.Errorf("Error setting `identity`: %+v", err)
 	}
 
-	d.Set("vm_size", resp.VirtualMachineProperties.HardwareProfile.VMSize)
-
-	if resp.VirtualMachineProperties.StorageProfile.ImageReference != nil {
-		if err := d.Set("storage_image_reference", schema.NewSet(resourceArmVirtualMachineStorageImageReferenceHash, flattenAzureRmVirtualMachineImageReference(resp.VirtualMachineProperties.StorageProfile.ImageReference))); err != nil {
-			return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage Image Reference error: %#v", err)
+	if props := resp.VirtualMachineProperties; props != nil {
+		if availabilitySet := props.AvailabilitySet; availabilitySet != nil {
+			// TODO: why is this being lower-cased?
+			d.Set("availability_set_id", strings.ToLower(*availabilitySet.ID))
 		}
-	}
 
-	if osDisk := resp.VirtualMachineProperties.StorageProfile.OsDisk; osDisk != nil {
-		diskInfo, err := resourceArmVirtualMachineGetManagedDiskInfo(osDisk.ManagedDisk, meta)
-		if err != nil {
-			return fmt.Errorf("[DEBUG] Error getting managed OS disk detailed information: %#v", err)
+		if profile := props.HardwareProfile; profile != nil {
+			d.Set("vm_size", profile.VMSize)
 		}
-		if err := d.Set("storage_os_disk", flattenAzureRmVirtualMachineOsDisk(osDisk, diskInfo)); err != nil {
-			return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage OS Disk error: %#v", err)
-		}
-	}
 
-	if dataDisks := resp.VirtualMachineProperties.StorageProfile.DataDisks; dataDisks != nil {
-		disksInfo := make([]*compute.Disk, len(*dataDisks))
-		for i, dataDisk := range *dataDisks {
-			diskInfo, err := resourceArmVirtualMachineGetManagedDiskInfo(dataDisk.ManagedDisk, meta)
-			if err != nil {
-				return fmt.Errorf("[DEBUG] Error getting managed data disk detailed information: %#v", err)
+		if profile := props.StorageProfile; profile != nil {
+			if err := d.Set("storage_image_reference", schema.NewSet(resourceArmVirtualMachineStorageImageReferenceHash, flattenAzureRmVirtualMachineImageReference(profile.ImageReference))); err != nil {
+				return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage Image Reference error: %#v", err)
 			}
-			disksInfo[i] = diskInfo
-		}
-		if err := d.Set("storage_data_disk", flattenAzureRmVirtualMachineDataDisk(dataDisks, disksInfo)); err != nil {
-			return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage Data Disks error: %#v", err)
-		}
-	}
 
-	if resp.VirtualMachineProperties.OsProfile != nil {
-		if err := d.Set("os_profile", schema.NewSet(resourceArmVirtualMachineStorageOsProfileHash, flattenAzureRmVirtualMachineOsProfile(resp.VirtualMachineProperties.OsProfile))); err != nil {
-			return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage OS Profile: %#v", err)
-		}
+			if osDisk := profile.OsDisk; osDisk != nil {
+				diskInfo, err := resourceArmVirtualMachineGetManagedDiskInfo(osDisk.ManagedDisk, meta)
+				if err != nil {
+					return fmt.Errorf("Error flattening `storage_os_disk`: %#v", err)
+				}
+				if err := d.Set("storage_os_disk", flattenAzureRmVirtualMachineOsDisk(osDisk, diskInfo)); err != nil {
+					return fmt.Errorf("Error setting `storage_os_disk`: %#v", err)
+				}
+			}
 
-		if resp.VirtualMachineProperties.OsProfile.WindowsConfiguration != nil {
-			if err := d.Set("os_profile_windows_config", schema.NewSet(resourceArmVirtualMachineStorageOsProfileWindowsConfigHash, flattenAzureRmVirtualMachineOsProfileWindowsConfiguration(resp.VirtualMachineProperties.OsProfile.WindowsConfiguration))); err != nil {
-				return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage OS Profile Windows Configuration: %#v", err)
+			if dataDisks := profile.DataDisks; dataDisks != nil {
+				disksInfo := make([]*compute.Disk, len(*dataDisks))
+				for i, dataDisk := range *dataDisks {
+					diskInfo, err := resourceArmVirtualMachineGetManagedDiskInfo(dataDisk.ManagedDisk, meta)
+					if err != nil {
+						return fmt.Errorf("[DEBUG] Error getting managed data disk detailed information: %#v", err)
+					}
+					disksInfo[i] = diskInfo
+				}
+				if err := d.Set("storage_data_disk", flattenAzureRmVirtualMachineDataDisk(dataDisks, disksInfo)); err != nil {
+					return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage Data Disks error: %#v", err)
+				}
 			}
 		}
 
-		if resp.VirtualMachineProperties.OsProfile.LinuxConfiguration != nil {
-			if err := d.Set("os_profile_linux_config", schema.NewSet(resourceArmVirtualMachineStorageOsProfileLinuxConfigHash, flattenAzureRmVirtualMachineOsProfileLinuxConfiguration(resp.VirtualMachineProperties.OsProfile.LinuxConfiguration))); err != nil {
-				return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage OS Profile Linux Configuration: %#v", err)
+		if profile := props.OsProfile; profile != nil {
+			if err := d.Set("os_profile", schema.NewSet(resourceArmVirtualMachineStorageOsProfileHash, flattenAzureRmVirtualMachineOsProfile(profile))); err != nil {
+				return fmt.Errorf("Error setting `os_profile`: %#v", err)
+			}
+
+			if err := d.Set("os_profile_linux_config", schema.NewSet(resourceArmVirtualMachineStorageOsProfileLinuxConfigHash, flattenAzureRmVirtualMachineOsProfileLinuxConfiguration(profile.LinuxConfiguration))); err != nil {
+				return fmt.Errorf("Error setting `os_profile_linux_config`: %+v", err)
+			}
+
+			if err := d.Set("os_profile_windows_config", schema.NewSet(resourceArmVirtualMachineStorageOsProfileWindowsConfigHash, flattenAzureRmVirtualMachineOsProfileWindowsConfiguration(profile.WindowsConfiguration))); err != nil {
+				return fmt.Errorf("Error setting `os_profile_windows_config`: %+v", err)
+			}
+
+			if err := d.Set("os_profile_secrets", flattenAzureRmVirtualMachineOsProfileSecrets(profile.Secrets)); err != nil {
+				return fmt.Errorf("Error setting `os_profile_secrets`: %+v", err)
 			}
 		}
 
-		if resp.VirtualMachineProperties.OsProfile.Secrets != nil {
-			if err := d.Set("os_profile_secrets", flattenAzureRmVirtualMachineOsProfileSecrets(resp.VirtualMachineProperties.OsProfile.Secrets)); err != nil {
-				return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage OS Profile Secrets: %#v", err)
+		if profile := props.DiagnosticsProfile; profile != nil {
+			if err := d.Set("boot_diagnostics", flattenAzureRmVirtualMachineDiagnosticsProfile(profile.BootDiagnostics)); err != nil {
+				return fmt.Errorf("Error setting `boot_diagnostics`: %#v", err)
 			}
 		}
-	}
 
-	if resp.VirtualMachineProperties.DiagnosticsProfile != nil && resp.VirtualMachineProperties.DiagnosticsProfile.BootDiagnostics != nil {
-		if err := d.Set("boot_diagnostics", flattenAzureRmVirtualMachineDiagnosticsProfile(resp.VirtualMachineProperties.DiagnosticsProfile.BootDiagnostics)); err != nil {
-			return fmt.Errorf("[DEBUG] Error setting Virtual Machine Diagnostics Profile: %#v", err)
-		}
-	}
+		if profile := props.NetworkProfile; profile != nil {
+			if err := d.Set("network_interface_ids", flattenAzureRmVirtualMachineNetworkInterfaces(profile)); err != nil {
+				return fmt.Errorf("Error flattening `network_interface_ids`: %#v", err)
+			}
 
-	if resp.VirtualMachineProperties.NetworkProfile != nil {
-		if err := d.Set("network_interface_ids", flattenAzureRmVirtualMachineNetworkInterfaces(resp.VirtualMachineProperties.NetworkProfile)); err != nil {
-			return fmt.Errorf("[DEBUG] Error setting Virtual Machine Storage Network Interfaces: %#v", err)
-		}
-
-		if resp.VirtualMachineProperties.NetworkProfile.NetworkInterfaces != nil {
-			for _, nic := range *resp.VirtualMachineProperties.NetworkProfile.NetworkInterfaces {
-				if props := nic.NetworkInterfaceReferenceProperties; props != nil {
-					if props.Primary != nil && *props.Primary {
-						d.Set("primary_network_interface_id", nic.ID)
-						break
+			if profile.NetworkInterfaces != nil {
+				for _, nic := range *profile.NetworkInterfaces {
+					if props := nic.NetworkInterfaceReferenceProperties; props != nil {
+						if props.Primary != nil && *props.Primary {
+							d.Set("primary_network_interface_id", nic.ID)
+							break
+						}
 					}
 				}
 			}
@@ -967,15 +966,30 @@ func resourceArmVirtualMachineDeleteManagedDisk(managedDiskID string, meta inter
 }
 
 func flattenAzureRmVirtualMachinePlan(plan *compute.Plan) []interface{} {
+	if plan == nil {
+		return []interface{}{}
+	}
+
 	result := make(map[string]interface{})
-	result["name"] = *plan.Name
-	result["publisher"] = *plan.Publisher
-	result["product"] = *plan.Product
+
+	if plan.Name != nil {
+		result["name"] = *plan.Name
+	}
+	if plan.Publisher != nil {
+		result["publisher"] = *plan.Publisher
+	}
+	if plan.Product != nil {
+		result["product"] = *plan.Product
+	}
 
 	return []interface{}{result}
 }
 
 func flattenAzureRmVirtualMachineImageReference(image *compute.ImageReference) []interface{} {
+	if image == nil {
+		return []interface{}{}
+	}
+
 	result := make(map[string]interface{})
 	if image.Publisher != nil {
 		result["publisher"] = *image.Publisher
@@ -1027,9 +1041,16 @@ func flattenAzureRmVirtualMachineIdentity(identity *compute.VirtualMachineIdenti
 }
 
 func flattenAzureRmVirtualMachineDiagnosticsProfile(profile *compute.BootDiagnostics) []interface{} {
+	if profile == nil {
+		return []interface{}{}
+	}
+
 	result := make(map[string]interface{})
 
-	result["enabled"] = *profile.Enabled
+	if profile.Enabled != nil {
+		result["enabled"] = *profile.Enabled
+	}
+
 	if profile.StorageURI != nil {
 		result["storage_uri"] = *profile.StorageURI
 	}
@@ -1046,6 +1067,10 @@ func flattenAzureRmVirtualMachineNetworkInterfaces(profile *compute.NetworkProfi
 }
 
 func flattenAzureRmVirtualMachineOsProfileSecrets(secrets *[]compute.VaultSecretGroup) []interface{} {
+	if secrets == nil {
+		return []interface{}{}
+	}
+
 	result := make([]interface{}, 0)
 	for _, secret := range *secrets {
 		s := map[string]interface{}{
@@ -1117,6 +1142,10 @@ func flattenAzureRmVirtualMachineOsProfile(input *compute.OSProfile) []interface
 }
 
 func flattenAzureRmVirtualMachineOsProfileWindowsConfiguration(config *compute.WindowsConfiguration) []interface{} {
+	if config == nil {
+		return []interface{}{}
+	}
+
 	result := make(map[string]interface{})
 
 	if config.ProvisionVMAgent != nil {
@@ -1131,11 +1160,11 @@ func flattenAzureRmVirtualMachineOsProfileWindowsConfiguration(config *compute.W
 		result["timezone"] = *config.TimeZone
 	}
 
-	if config.WinRM != nil {
-		listeners := make([]map[string]interface{}, 0, len(*config.WinRM.Listeners))
+	listeners := make([]map[string]interface{}, 0)
+	if config.WinRM != nil && config.WinRM.Listeners != nil {
 		for _, i := range *config.WinRM.Listeners {
 			listener := make(map[string]interface{})
-			listener["protocol"] = i.Protocol
+			listener["protocol"] = string(i.Protocol)
 
 			if i.CertificateURL != nil {
 				listener["certificate_url"] = *i.CertificateURL
@@ -1143,17 +1172,17 @@ func flattenAzureRmVirtualMachineOsProfileWindowsConfiguration(config *compute.W
 
 			listeners = append(listeners, listener)
 		}
-
-		result["winrm"] = listeners
 	}
 
+	result["winrm"] = listeners
+
+	content := make([]map[string]interface{}, 0)
 	if config.AdditionalUnattendContent != nil {
-		content := make([]map[string]interface{}, 0, len(*config.AdditionalUnattendContent))
 		for _, i := range *config.AdditionalUnattendContent {
 			c := make(map[string]interface{})
-			c["pass"] = i.PassName
-			c["component"] = i.ComponentName
-			c["setting_name"] = i.SettingName
+			c["pass"] = string(i.PassName)
+			c["component"] = string(i.ComponentName)
+			c["setting_name"] = string(i.SettingName)
 
 			if i.Content != nil {
 				c["content"] = *i.Content
@@ -1161,23 +1190,31 @@ func flattenAzureRmVirtualMachineOsProfileWindowsConfiguration(config *compute.W
 
 			content = append(content, c)
 		}
-
-		result["additional_unattend_config"] = content
 	}
+	result["additional_unattend_config"] = content
 
 	return []interface{}{result}
 }
 
 func flattenAzureRmVirtualMachineOsProfileLinuxConfiguration(config *compute.LinuxConfiguration) []interface{} {
+	if config == nil {
+		return []interface{}{}
+	}
 
 	result := make(map[string]interface{})
-	result["disable_password_authentication"] = *config.DisablePasswordAuthentication
 
-	if config.SSH != nil && len(*config.SSH.PublicKeys) > 0 {
-		ssh_keys := make([]map[string]interface{}, 0, len(*config.SSH.PublicKeys))
+	if config.DisablePasswordAuthentication != nil {
+		result["disable_password_authentication"] = *config.DisablePasswordAuthentication
+	}
+
+	if config.SSH != nil && config.SSH.PublicKeys != nil && len(*config.SSH.PublicKeys) > 0 {
+		ssh_keys := make([]map[string]interface{}, 0)
 		for _, i := range *config.SSH.PublicKeys {
 			key := make(map[string]interface{})
-			key["path"] = *i.Path
+
+			if i.Path != nil {
+				key["path"] = *i.Path
+			}
 
 			if i.KeyData != nil {
 				key["key_data"] = *i.KeyData
@@ -1194,8 +1231,10 @@ func flattenAzureRmVirtualMachineOsProfileLinuxConfiguration(config *compute.Lin
 
 func flattenAzureRmVirtualMachineOsDisk(disk *compute.OSDisk, diskInfo *compute.Disk) []interface{} {
 	result := make(map[string]interface{})
-	result["name"] = *disk.Name
-	if disk.Vhd != nil {
+	if disk.Name != nil {
+		result["name"] = *disk.Name
+	}
+	if disk.Vhd != nil && disk.Vhd.URI != nil {
 		result["vhd_uri"] = *disk.Vhd.URI
 	}
 	if disk.Image != nil && disk.Image.URI != nil {

